makefile.shade

```diff
@@ -203,6 +204,40 @@ var buildTarget = "compile"
         }
     }
 
+#cache-coherence
+    @{
+        var coherenceCacheDir = Path.Combine(BASE_DIR, Environment.GetEnvironmentVariable("COHERENCE_CACHE_DIR") ?? ".coherence-cache");
+        var dropsShare = Environment.GetEnvironmentVariable("ASPNETCI_DROPS_SHARE");
+
+        if (!string.IsNullOrWhiteSpace(dropsShare))
+        {
+            var coherenceShare = Path.Combine(dropsShare, "Coherence", BUILD_BRANCH);
+            var latestBuild = Directory.EnumerateDirectories(coherenceShare)
+                .Select(d => Path.GetFileName(d))
+                .Where(d => { int _; return int.TryParse(d, out _); })
+                .OrderByDescending(d => d)
+                .First();
+            var targetDir = Path.Combine(coherenceCacheDir, latestBuild);
+            
+            Log.Info("Latest Coherence build is " + latestBuild     + ".");
+
+            if (!Directory.Exists(targetDir))
+            {
+                var latestBuildShare = Path.Combine(coherenceShare, latestBuild);
+
+                Log.Info("Caching Coherence build " + latestBuild + " at " + targetDir + ".");
+                CopyFolder(latestBuildShare, targetDir, overwrite: true);
+            }
+            else
+            {
+                Log.Info("Coherence build " + latestBuild + " already cached at " + targetDir + ".");
+            }
+            
+            Environment.SetEnvironmentVariable("NUGET_VOLATILE_FEED_ASPNETVNEXT", Path.Combine(targetDir, "build"));
+            Environment.SetEnvironmentVariable("NUGET_VOLATILE_FEED_EXTERNAL", Path.Combine(dropsShare, "latest-packages", "external", BUILD_BRANCH));
```

Comment:
There should be a way to overwrite this with an env var but we should move away from configurations that don't have defaults in code because it makes it hard to reproduce builds locally.

Btw, where are this variable and the one above used?